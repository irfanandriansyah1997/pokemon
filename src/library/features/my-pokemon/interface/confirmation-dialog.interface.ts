import {
  IDialogProps,
  IEventClose
} from '@/library/component/dialog/interface';
import { IEventComponent } from '@/library/interface/general';

export type IConfirmationEventSubmit = IEventComponent<'on-delete'>;

export type IConfirmationEvent = (
  param: IEventClose | IConfirmationEventSubmit
) => void;

/**
 * Confirmation Dialog Props
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.08.02
 */
export interface IConfirmationDialog extends Pick<IDialogProps, 'show'> {
  error?: string;
  name?: string;
  on: IConfirmationEvent;
}
