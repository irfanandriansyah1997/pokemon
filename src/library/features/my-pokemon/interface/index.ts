export * from './my-pokemon-hooks.interface';
export * from './register-dialog.interface';
