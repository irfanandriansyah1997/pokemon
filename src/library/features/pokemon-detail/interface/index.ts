export * from './pokemon-item.interface';
