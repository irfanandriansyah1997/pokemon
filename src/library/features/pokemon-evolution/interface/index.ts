export * from './pokemon-evolution.interface';
