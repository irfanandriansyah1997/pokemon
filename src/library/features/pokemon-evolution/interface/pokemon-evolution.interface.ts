/**
 * Evolution Label Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.08.01
 */
export interface IEvolutionItem {
  imageAfter: IEvolutionItemPokemon;
  imageBefore: IEvolutionItemPokemon;
  key: string;
  label: IEvolutionItemLabel;
}

/**
 * Evolution Item Label Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.08.01
 */
export interface IEvolutionItemLabel {
  attributes: string;
  name: string;
}

/**
 * Evolution Item Pokemon Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.08.01
 */
export interface IEvolutionItemPokemon {
  images: string;
  pokeName: string;
}
