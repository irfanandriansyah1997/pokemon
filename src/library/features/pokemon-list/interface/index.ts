export * from './pokemon-list-hooks.interface';
export * from './pokemon-card.interface';
