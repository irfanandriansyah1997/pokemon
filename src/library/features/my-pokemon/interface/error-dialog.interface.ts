import { IDialogProps } from '@/library/component/dialog/interface';

export type IErrorDialogProps = Pick<IDialogProps, 'on' | 'show'>;
