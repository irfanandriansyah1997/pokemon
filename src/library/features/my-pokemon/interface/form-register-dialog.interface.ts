import {
  IDialogProps,
  IEventClose
} from '@/library/component/dialog/interface';
import { IEventComponent } from '@/library/interface/general';

export type IRegisterEventSubmit = IEventComponent<'on-submit', string>;

export type IFormRegisterEvent = (
  param: IEventClose | IRegisterEventSubmit
) => void;

/**
 * Form Register Dialog
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.08.02
 */
export interface IFormRegisterDialog extends Pick<IDialogProps, 'show'> {
  error?: string;
  on: IFormRegisterEvent;
}
