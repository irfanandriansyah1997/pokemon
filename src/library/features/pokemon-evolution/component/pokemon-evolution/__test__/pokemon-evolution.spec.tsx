import { MockedProvider } from '@apollo/client/testing';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

import { MOCK_GRAPHQL_RESPONSE } from '@/library/__mock__/graphql-response.mock';
import PokemonFixture from '@/library/features/pokemon-detail/reducers/__test__/__fixture__/full-pokemon.fixture.json';
import PokemonEvolution from '@/library/features/pokemon-evolution/component/pokemon-evolution';
import { IPokemon } from '@/library/interface/pokemon';

describe(`Testing Evolution Component`, () => {
  it(`Renders properly with default configuration`, async () => {
    const ui = mount(
      <MockedProvider mocks={MOCK_GRAPHQL_RESPONSE}>
        <PokemonEvolution {...((PokemonFixture as unknown) as IPokemon)} />
      </MockedProvider>
    );

    expect(ui.html()).toMatchSnapshot();

    await act(
      () =>
        new Promise<void>((resolve) => {
          setTimeout(() => {
            ui.update();
            resolve();
          }, 1000);
        })
    );

    expect(ui.html()).toMatchSnapshot();
  });
});
