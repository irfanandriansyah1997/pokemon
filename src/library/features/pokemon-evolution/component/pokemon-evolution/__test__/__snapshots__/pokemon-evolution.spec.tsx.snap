// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Evolution Component Renders properly with default configuration 1`] = `"<p font-weight=\\"500\\" font-size=\\"medium\\" class=\\"css-14e7f3w\\"><br>Loading</p>"`;

exports[`Testing Evolution Component Renders properly with default configuration 2`] = `"<p class=\\"css-1e7cmzx\\">Evolution bulbasaur</p><div class=\\"css-2gx4tk\\"><div class=\\"css-177dxd8\\"><img loading=\\"lazy\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png\\" alt=\\"bulbasaur\\"><p font-weight=\\"500\\" font-size=\\"text\\" class=\\"css-17tivp9\\">bulbasaur</p></div><div class=\\"css-xr71pc\\"><img loading=\\"lazy\\" src=\\"next-arrow.svg\\" alt=\\"evolution arrow\\"><p font-weight=\\"500\\" font-size=\\"normal\\" color=\\"primary\\" class=\\"css-nftjgg\\">level-up</p><p font-weight=\\"400\\" font-size=\\"text\\" color=\\"secondary\\" class=\\"css-1brd6hq\\">Min Level 16</p></div><div class=\\"css-177dxd8\\"><img loading=\\"lazy\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png\\" alt=\\"ivysaur\\"><p font-weight=\\"500\\" font-size=\\"text\\" class=\\"css-17tivp9\\">ivysaur</p></div></div><div class=\\"css-2gx4tk\\"><div class=\\"css-177dxd8\\"><img loading=\\"lazy\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png\\" alt=\\"ivysaur\\"><p font-weight=\\"500\\" font-size=\\"text\\" class=\\"css-17tivp9\\">ivysaur</p></div><div class=\\"css-xr71pc\\"><img loading=\\"lazy\\" src=\\"next-arrow.svg\\" alt=\\"evolution arrow\\"><p font-weight=\\"500\\" font-size=\\"normal\\" color=\\"primary\\" class=\\"css-nftjgg\\">level-up</p><p font-weight=\\"400\\" font-size=\\"text\\" color=\\"secondary\\" class=\\"css-1brd6hq\\">Min Level 32</p></div><div class=\\"css-177dxd8\\"><img loading=\\"lazy\\" src=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png\\" alt=\\"venusaur\\"><p font-weight=\\"500\\" font-size=\\"text\\" class=\\"css-17tivp9\\">venusaur</p></div></div>"`;
